internal/handlers/handler.go

```diff
@@ -0,0 +1,120 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/charmbracelet/log"
+	"github.com/labstack/echo/v4"
+	"github.com/rackerlabs/external-dns-rackspace-webhook/internal/providers"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+)
+
+type Handler struct {
+	provider *providers.RackspaceProvider
+}
+
+func NewHandler(rackspaceProvider *providers.RackspaceProvider) *Handler {
+	return &Handler{
+		provider: rackspaceProvider,
+	}
+}
+
+func (h *Handler) NegotiationHandler(c echo.Context) error {
+	c.Response().Header().Set(echo.HeaderContentType, "application/external.dns.webhook+json;version=1")
+	return c.JSON(http.StatusOK, h.provider.DomainFilter)
+}
+
+func (h *Handler) HandleGetRecords(c echo.Context) error {
+	endpoints, err := h.provider.Records(c.Request().Context())
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, map[string]string{"error": "Failed to fetch records"})
```

Comment:
can we include actual error or response body returned from provider here? It would be difficult to troubleshoot with only generic 'Failed to fetch records'